### DISCUSSION TITLE
Opt-out module if we don't need it

### DISCUSSION THREAD

**Discussion by thanhlmm** [TanStack/query#3150] [Q&A]
Hi, we have used `react-query` and it's fantastic tool ever created.

After I checked the bundle packages, I saw some modules maybe we can opt-out because I don't actually use it in the project. So my question is, can we have a way to opt-out those modules?

<img width="1044" alt="Screen Shot 2021-12-30 at 5 21 50 PM" src="https://user-images.githubusercontent.com/9281080/147743158-f2799c24-9737-4e0b-8bdd-30ac981370f6.png">

**Comment by TkDodo** [comment]
those are mostly internal module that are used implicitly.

**Reply by thanhlmm** [reply]
My used case is, I don't want to care about `refetchOnReconnect` and onlineManager. So I want to opt it out

**Reply by thanhlmm** [reply]
I also sê that Observer is rarely have use cases

**Reply by TkDodo** [reply]
Every useQuery call creates an observer...
